### PR TITLE
feat: add support to customize gcloud command of LocalDatastoreHelper

### LINKED ISSUE #19: Customize gcloudCommand of LocalDatastoreHelper
I am using Spring Cloud GCP. It has the auto-configuration to start datastore emulator when starting the app engine devserver. However, when I read through the set up of LocalDatastoreHelper, I could see that the gcloudCommand is hard-coded. Are there any ways that I can customize the options of gcloudCommand ?

```
    List<String> gcloudCommand = new ArrayList<>(Arrays.asList(GCLOUD_CMD_TEXT.split(" ")));
    gcloudCommand.add(GCLOUD_CMD_PORT_FLAG + "localhost:" + getPort());
    gcloudCommand.add(CONSISTENCY_FLAG + consistency);
    gcloudCommand.add("--no-store-on-disk");
    GcloudEmulatorRunner gcloudRunner =
        new GcloudEmulatorRunner(gcloudCommand, VERSION_PREFIX, MIN_VERSION);
    if (gcdPath != null) {
      gcloudCommand.add("--data-dir=" + gcdPath.toString());
    }
```

I want to remove the option `--not-store-on-disk` and would like to change the `--data-dir` to my local folder.
Moreover, could you help me to find the location of the current data-dir. I do not know where it is.
Thanks

### PR DESCRIPTION
**Pull request by athakor**
Fixes #19

### CODE REVIEW

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java [hunk 1]
```diff
@@ -43,6 +45,7 @@
   private static final double TOLERANCE = 0.00001;
   private static final String PROJECT_ID_PREFIX = "test-project-";
   private static final String NAMESPACE = "namespace";
+  private static final Path DATA_DIR = Paths.get("DATA-DIR");
```
[the remark this answers is not in the document]
Comment (athakor) [reply google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java]: done

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java [hunk 2]
```diff
@@ -54,6 +57,56 @@ public void testCreate() {
...
+            .setStoreOnDisk(false)
+            .setDataDir(DATA_DIR)
+            .build();
+    assertTrue(Math.abs(0.75 - helper.getConsistency()) < TOLERANCE);
+    assertTrue(helper.getProjectId().startsWith(PROJECT_ID_PREFIX));
+    assertFalse(helper.isStoreOnDisk());
+    assertEquals(8081, helper.getPort());
+    assertEquals(DATA_DIR, helper.getGcdPath());
+    LocalDatastoreHelper actualHelper = helper.toBuilder().build();
+    compareLocalDatastoreHelper(helper, actualHelper);
```
[the remark this answers is not in the document]
Comment (athakor) [reply google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java]: done

**Comment by athakor** [comment]
> This does not build & test cleanly for me in my environment which doesn't have `gcloud` on my `PATH`, the full test suite would run successfully prior to this change.
> 
> My java & maven:
> 
> ```
> Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T11:06:16-04:00)
> Maven home: ~/opt/maven/latest
> Java version: 1.8.0_201, vendor: Oracle Corporation, runtime: ~/opt/java/jdk1.8.0_201/jre
> Default locale: en_US, platform encoding: UTF-8
> OS name: "linux", version: "5.2.17-1rodete3-amd64", arch: "amd64", family: "unix"
> ```
> 
> build log with timeout failure

@BenWhitehead Thanks for catching this, it's fixed please check again this in your env.

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java [hunk 3]
```diff
@@ -43,6 +46,12 @@
   private static final double TOLERANCE = 0.00001;
   private static final String PROJECT_ID_PREFIX = "test-project-";
   private static final String NAMESPACE = "namespace";
+  private Path dataDir;
+
+  @Before
+  public void setUp() throws IOException {
+    dataDir = Files.createTempDirectory("gcd");
```
Comment (BenWhitehead) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java:53]: Looks good, can you also add a delete in an `@After` so we're not leaving lots of directories around?

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java [hunk 4]
```diff
@@ -54,6 +63,56 @@ public void testCreate() {
...
+            .setDataDir(dataDir)
+            .build();
+    assertTrue(Math.abs(0.75 - helper.getConsistency()) < TOLERANCE);
+    assertTrue(helper.getProjectId().startsWith(PROJECT_ID_PREFIX));
+    assertFalse(helper.isStoreOnDisk());
+    assertEquals(8081, helper.getPort());
+    assertEquals(dataDir, helper.getGcdPath());
+    LocalDatastoreHelper actualHelper = helper.toBuilder().build();
+    assertLocalDatastoreHelpersEquivelent(helper, actualHelper);
+    Path dataDir = Files.createTempDirectory("gcd_data_dir");
```
Comment (BenWhitehead) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java:101]: Cleanup dir like above

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java (hunk 3 above)
Comment (athakor) [reply google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java:54]: done

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java (hunk 4 above)
Comment (athakor) [reply google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java:107]: done

**Comment by athakor** [comment]
@BenWhitehead  Both the comments have been addressed. PTAL

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java [hunk 5]
```diff
@@ -53,6 +54,11 @@ public void setUp() throws IOException {
     dataDir = Files.createTempDirectory("gcd");
   }
 
+  @After
+  public void tearDown() throws IOException {
+    Files.delete(dataDir);
```
Comment (BenWhitehead) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java:59]: Calling `Files.delete` will fail if it is a directory which is not empty [javadoc](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#delete-java.nio.file.Path-).

Instead you can use the helper method  com.google.cloud.datastore.testing.LocalDatastoreHelper#deleteRecursively by making it package-private.

Comment (athakor) [reply google-cloud-datastore/src/test/java/com/google/cloud/datastore/testing/ITLocalDatastoreHelperTest.java]: done

**Pull request merged**